django/urls/resolvers.py

```diff
@@ -151,48 +139,218 @@ def _check_pattern_startswith_slash(self):
         else:
             return []
 
+    def _check_include_trailing_dollar(self):
+        """
+        Check that include is not used with a regex ending with a dollar.
+        """
+        regex_pattern = self.regex.pattern
+        if regex_pattern.endswith('$') and not regex_pattern.endswith(r'\$'):
+            warning = Warning(
+                "Your URL pattern {} uses include with a regex ending with a '$'. "
+                "Remove the dollar from the regex to avoid problems including "
+                "URLs.".format(self.describe()),
+                id="urls.W001",
+            )
+            return [warning]
+        else:
+            return []
+
+
+class RegexPattern(CheckURLMixin):
+    regex = LocaleRegexDescriptor('_regex')
+
+    def __init__(self, regex, name=None, is_endpoint=False, converters=None):
+        self._regex = regex
+        self._regex_dict = {}
+        self._is_endpoint = is_endpoint
+        self.name = name
+        self.converters = converters or {}
+
+    def match(self, path):
+        match = self.regex.search(path)
+        if match:
+            # If there are any named groups, use those as kwargs, ignoring
+            # non-named groups. Otherwise, pass all non-named arguments as
+            # positional arguments.
+            kwargs = match.groupdict()
+            args = () if kwargs else match.groups()
+            # In both cases, pass any extra_kwargs as **kwargs.
+
+            for key, value in kwargs.items():
```

Comment:
The `converters` argument for `re_path()` is currently unused and untested and therefore this loop does nothing since `self.converters` is always empty. Is it supposed to be a documented feature or can it be removed?

Reply:
I think keeping it makes sense. That way re_path at least supports everything the normal path() has to offer, albeit with a less friendly syntax.

Reply:
Do you have a use case in mind we could use for the documentation?

Reply:
Unfortunately, no. But I also don't really have a good use case for when `re_path` is even necessary anymore. It was more of a theoretical concern: what if you have a view that uses `path()` for most (with converters), but for one particular use-case you use `re_path` but still would like the converters to work.

Reply:
How about omitting it until we have a use case? That will save writing tests and docs for a theoretical feature. :-)
From a readability point of view, writing a `re_path()` that mixes regexes and converters in the string, and then has to initialize and pass converters in the URLconf sounds nasty and not something to encourage!

Reply:
I'm going to be a +1 to just dropping `converters`

Reply:
Agreed.